### PR TITLE
Local JWT settings

### PR DESCRIPTION
**Pull request by mushtaqak**
* Adds CORS library to handle cross domain requests to service. 
* Adds the port from our development server to the local whitelist. 
* Adds `JWT_PUBLIC_SIGNING_JWK_SET` JWT auth settings for local development.

### CODE REVIEW

**Comment by mushtaqak** [comment]
@georgebabey Can you have a look?

File: ecommerce/settings/local.py [hunk 1]
```diff
@@ -53,7 +55,7 @@
         {
             'SECRET_KEY': 'lms-secret',
             'AUDIENCE': 'lms-key',
-            'ISSUER': 'http://edx.devstack.lms:18000/oauth2'
+            'ISSUER': 'http://127.0.0.1:8000/oauth2'
```
Comment (mushtaqak) [inline ecommerce/settings/local.py]: Note to reviewers: Ignore this, I will remove this - I still use vagrant ;)

File: requirements/base.in [hunk 2]
```diff
@@ -8,6 +8,7 @@
 coreapi
 django==1.11.15
 django-compressor==2.2
+django-cors-headers                 #  Used to allow to configure CORS headers for cross-domain requests
```
Comment (mushtaqak) [inline requirements/base.in:11]: I will also have to add .txt requirement files.

File: ecommerce/settings/local.py [hunk 3]
```diff
@@ -67,7 +69,22 @@
...
+    'JWT_PUBLIC_SIGNING_JWK_SET': (
+        '{"keys": [{"kid": "devstack_key", "e": "AQAB", "kty": "RSA", "n": "smKFSYowG6nNUAdeqH1jQQnH1PmIHphzBmwJ5vRf1vu'
+        '48BUI5VcVtUWIPqzRK_LDSlZYh9D0YFL0ZTxIrlb6Tn3Xz7pYvpIAeYuQv3_H5p8tbz7Fb8r63c1828wXPITVTv8f7oxx5W3lFFgpFAyYMmROC'
+        '4Ee9qG5T38LFe8_oAuFCEntimWxN9F3P-FJQy43TL7wG54WodgiM0EgzkeLr5K6cDnyckWjTuZbWI-4ffcTgTZsL_Kq1owa_J2ngEfxMCObnzG'
+        'y5ZLcTUomo4rZLjghVpq6KZxfS6I1Vz79ZsMVUWEdXOYePCKKsrQG20ogQEkmTf9FT_SouC6jPcHLXw"}]}'
+    ),
 })
+
+CORS_ORIGIN_WHITELIST = (
+    'localhost:1991'
```
Comment (robrap) [inline ecommerce/settings/local.py:81]: FYI: When I first saw this, I was curious why gradebook needed to call ecommerce.  This note is tangentially related to this PR.  We should probably find a way to document/track our port usage to keep it unique across our devstack.  Here is where we track the ports of [everything that comes with devstack](https://github.com/edx/devstack#service-urls).  Maybe another column to say whether it is included?

Comment (georgebabey) [reply ecommerce/settings/local.py:81]: @robrap do you know if the django cookiecutter sets you up with a unique port? We should probably find a way to do that with the frontend cookiecutter so everything doesn't end up being on 1991

File: ecommerce/settings/local.py (hunk 1 above)
Comment (georgebabey) [reply ecommerce/settings/local.py]: shame.... 😄

File: ecommerce/settings/local.py (hunk 3 above)
Comment (robrap) [reply ecommerce/settings/local.py:81]: For the IDA cookiecutter, it looks like the [port is based on a setup variable](https://github.com/edx/cookiecutter-django-ida/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/docker-compose.yml#L25).

Comment (georgebabey) [reply ecommerce/settings/local.py:81]: Thanks. We also need to get these microfrontends into devstack. I believe everyone right now is just running them locally - I'll mention it at the FedX meeting tomorrow

**Pull request merged**